src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -160,6 +160,11 @@ public void removeAutomaticResults() {
         this.results = this.results.stream().filter(result -> result == null || !result.isAutomatic()).collect(Collectors.toList());
     }
 
+    @JsonIgnore
+    public void removeNullResults() {
+        this.results = this.results.stream().filter(result -> result != null).collect(Collectors.toList());
```

Comment:
As far as I know, I think it has to do with second correction rounds

Reply:
Yes it can happen when retrieving the submissions from the database all results that have not been assessed by the calling user will be null values as the don't match the query, but hibernate has to keep the ordering based on the Order_Column